Tasks/SqlAzureDacpacDeployment/Utility.ps1

```diff
@@ -147,10 +147,19 @@ function Get-SqlPackageCommandArguments
 
         if($sqlUsername)
         {
-            if((-not $sqlUsername.Trim().EndsWith("@" + $serverName.Trim())) -and $sqlUsername.Contains('@'))
+            if ($serverName)
             {
-                $sqlUsername = $sqlUsername + "@" + $serverName 
+                $serverNameSplittedArgs = $serverName.Trim().Split(".")
+                if ($serverNameSplittedArgs.Length -gt 0)
+                {
```

Comment:
Can you add L0 for all the combinations; with @, without @, server name with and without FQDN and server name with port.